### PR TITLE
feat: add tree-shakability and update project structure

### LINKED ISSUE #10: Production build imports all icons !
i am only loading the following

```js
import {
  PhLockKeyOpen,
  PhUser,
  PhSun,
  PhMoon,
  PhPlayCircle,
  PhImage,
} from 'phosphor-vue'

import Vue from 'vue'

Vue.component('PhLockKeyOpen', PhLockKeyOpen)
Vue.component('PhUser', PhUser)
Vue.component('PhSun', PhSun)
Vue.component('PhMoon', PhMoon)
Vue.component('PhPlay', PhPlayCircle)
Vue.component('PhImage', PhImage)
```

the build result creates big javascript file

Code_q7Olh5tvui
msedge_MRLBAUkRV0

### LINKED ISSUE #15: Icons not tree-shakable by other bundlers
Don't know where the other issue went, so I'll create a new one to keep track of this.

It seems that with the current configuration of Rollup, it's not possible to tree shake unused icons when using this library with other bundlers such as Webpack.

### PR DESCRIPTION
**Pull request by dnlsndr**
This commit add tree-shakability so that one can import only the icons that one needs.
Due to some other improvements and general standards, the project structure has changed
a tiny bit.

fixes #15

### CODE REVIEW

**Comment by dnlsndr** [comment]
Seems like there is still an issue i need to fix. Damn, it's not as easy as it seems. The library is now compiled to esmodules, but it still seems as if it's not tree-shakable. Will have to do some more research.

**Comment by dnlsndr** [comment]
For anybody interested, here's a good resource on what's the issue at hand (mainly compiling with rollup, and tree-shaking with webpack as already stated in a different issue): https://blog.theodo.com/2021/04/library-tree-shaking/

**Comment by dnlsndr** [comment]
@rektdeckard I've fixed some final issues here. I created a package of this repo to test it in our local setups under `@dnlsndr/vue-phosphor-icons` and it works pretty well.

**Comment by rektdeckard** [comment]
Thanks for putting the work in @dnlsndr. Please run the formatting script to eliminate all these single quotes and properly indent all the generated code. I'll give this a test later tonight!

**Comment by rektdeckard** [comment]
For future ref, you can use the `example` demo app to test the bundled lib in a demo app immediately, or `npm link` command to test in an existing local project, without having to gum up the npm public registry with random test modules.

**Comment by dnlsndr** [comment]
@rektdeckard Yeah sorry about that. I do have an issue with using the example app, because it only uses the source files instead of testing the production output, so it's not really representative of what you may get in the end after building the library, especially when testing for tree-shakability. One could of courae just do a `npm pack` or as you mentioned `npm link` for testing but one wouldn't want to gunk up the repo with yet another special configuration in the `package.json` or is it possible with `npm link` without mutating the repo for that special testing case? (I'm really not all to versed in npm package deployment). I'm also not in favor of needlessly publishing packages to `npm`, but since we ourselves also needed a quick fix, I had to publish the fixed version of the package anyways for our deployments to reference. Maybe one could add a `CONTRIBUTING` guide so that anybody who wants to fix something with the library knows exactly what they need to do (including running the format script). I'd be glad to help out with this repo in any way possible, since as I already mentioned, we do also heavily rely on Phosphor icons at work and thus also want to ensure that the library meets general standards.

**Comment by dnlsndr** [comment]
I've reverted several needless formatting changes.

**Comment by dnlsndr** [comment]
I think it should be ready for review now

**Comment by rektdeckard** [comment]
Taking a look now!

File: bin/assemble.js [hunk 1]
```diff
@@ -200,6 +205,71 @@ export { default as Ph${name} } from "./Ph${name}.vue";
...
+      .split("-")
+      .map((substr) => substr.replace(/^\w/, (c) => c.toUpperCase()))
+      .join("");
+    indexString += `\nexport type Ph${name} = ExtendedVue<Vue, {}, {}, IconComputed, IconProps>;`;
+  }
+  try {
+    fs.writeFileSync(TYPES_PATH, indexString, {
+      flag: "w",
+    });
+    console.log(chalk.green("Export success"));
```
Comment (rektdeckard) [inline bin/assemble.js]:
```suggestion
    console.log(chalk.green("Typegen success"));
```

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
   "browser": "dist/esm/src/entry.esm.js",
   "module": "dist/esm/src/entry.esm.js",
   "unpkg": "dist/vue-phosphor-icons.min.js",
```
Comment (rektdeckard) [inline package.json]:
```suggestion
  "unpkg": "dist/phosphor-vue.min.js",
```

**Review by rektdeckard — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (dnlsndr) [reply package.json]: Don't know how that slipped in there

**Comment by rektdeckard** [comment]
nit: keep commit titles to 80 chars or move rest to description

**Comment by rektdeckard** [comment]
Works a charm, thanks @dnlsndr

**Comment by rektdeckard** [comment]
@dnlsndr Actually, reverting because it's incorrect to export the component types as a `.ts` file. We should emit ambient declarations in a `.d.ts` file using `export declare` syntax, in the way that Definitely Typed would do.

How we do it in the React lib: https://unpkg.com/browse/phosphor-react@1.4.0/dist/index.d.ts

**Comment by dnlsndr** [comment]
I'll have a look

**Pull request merged**